code/datums/elements/squashable.dm

```diff
@@ -11,57 +9,57 @@
 	///Special callback to call on squash instead, for things like hauberoach
 	var/datum/callback/on_squash_callback
 
-
-/datum/element/squashable/Attach(mob/living/target, squash_chance, squash_damage, squash_flags, squash_callback)
+/datum/component/squashable/Initialize(squash_chance, squash_damage, squash_flags, squash_callback)
 	. = ..()
-	if(!istype(target))
-		return ELEMENT_INCOMPATIBLE
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
 	if(squash_chance)
 		src.squash_chance = squash_chance
 	if(squash_damage)
 		src.squash_damage = squash_damage
 	if(squash_flags)
 		src.squash_flags = squash_flags
 	if(!src.on_squash_callback && squash_callback)
-		on_squash_callback = CALLBACK(target, squash_callback)
-
-	RegisterSignal(target, COMSIG_MOVABLE_CROSSED, .proc/OnCrossed)
+		on_squash_callback = CALLBACK(parent, squash_callback)
 
-/datum/element/squashable/Detach(mob/living/target)
-	UnregisterSignal(target, COMSIG_MOVABLE_CROSSED)
+	var/static/list/loc_connections = list(
+		COMSIG_MOVABLE_CROSSED = .proc/OnCrossed,
+	)
+	AddElement(/datum/element/connect_loc, parent, loc_connections)
```

Comment:
fixed top one and talked to you in discord about the bottom one. elements can no longer mutate the thing theyre attached to when registering to them being crossed because the signal only sends the turf and the atom crossing over